sdk/typescript/src/client/cryptography.ts

```diff
@@ -0,0 +1,358 @@
+		create_dkg_centralized_output(
+			networkDecryptionKeyPublicOutput,
+			Uint8Array.from(firstRoundOutput),
+			sessionIdentifierDigest(sessionIdentifier),
+		);
+
+	return {
+		centralizedPublicKeyShareAndProof: Uint8Array.from(centralizedPublicKeyShareAndProof),
+		centralizedPublicOutput: Uint8Array.from(centralizedPublicOutput),
+		centralizedSecretKeyShare: Uint8Array.from(centralizedSecretKeyShare),
+	};
+}
+
+/**
+ * Encrypt a secret share using the provided encryption key.
+ * This creates an encrypted share that can only be decrypted by the corresponding decryption key.
+ *
+ * @param centralizedSecretKeyShare - The secret key share to encrypt
+ * @param encryptionKey - The public encryption key to encrypt with
+ * @param networkDecryptionKeyPublicOutput - The network's public parameters for encryption
+ * @returns The encrypted secret share with proof of correct encryption
+ */
+export function encryptSecretShare(
+	centralizedSecretKeyShare: Uint8Array,
+	encryptionKey: Uint8Array,
+	networkDecryptionKeyPublicOutput: Uint8Array,
+): Uint8Array {
+	const encryptedUserShareAndProof = encrypt_secret_share(
+		centralizedSecretKeyShare,
+		encryptionKey,
+		networkDecryptionKeyPublicOutput,
+	);
+
+	return Uint8Array.from(encryptedUserShareAndProof);
+}
+
+/**
+ * Decrypt a user's encrypted secret share.
+ * This function verifies the encryption proof and decrypts the share using the private decryption key.
+ *
+ * @param decryptionKey - The private decryption key
+ * @param encryptionKey - The corresponding public encryption key
+ * @param dWalletOutput - The DWallet's public output for verification
+ * @param encryptedUserShareAndProof - The encrypted share with proof to decrypt
+ * @param networkDecryptionKeyPublicOutput - The network's public parameters
+ * @returns The decrypted secret share
+ * @throws {Error} If decryption fails or proof verification fails
+ */
+export function decryptUserShare(
+	decryptionKey: Uint8Array,
+	encryptionKey: Uint8Array,
+	dWalletOutput: Uint8Array,
+	encryptedUserShareAndProof: Uint8Array,
+	networkDecryptionKeyPublicOutput: Uint8Array,
+): Uint8Array {
+	const decryptedUserShare = decrypt_user_share(
+		decryptionKey,
+		encryptionKey,
+		dWalletOutput,
+		encryptedUserShareAndProof,
+		networkDecryptionKeyPublicOutput,
+	);
+
+	return Uint8Array.from(decryptedUserShare);
+}
+
+/**
+ * Prepare all cryptographic data needed for the second round of DKG.
+ * This function combines the DKG output generation and secret share encryption.
+ *
+ * @param networkDecryptionKeyPublicOutput - The network's public parameters
+ * @param dWallet - The DWallet object containing first round output
+ * @param sessionIdentifier - Unique identifier for this DKG session
+ * @param encryptionKey - The user's public encryption key
+ * @returns Complete prepared data for the second DKG round
+ * @throws {Error} If the first round output is not available in the DWallet
+ */
+export function prepareDKGSecondRound(
```

Comment:
Please separate the outputs:
* `dkg_second_round_request_data: DKGSecondRoundRequestData`
* `centralizedPartySecretKeyShare`
*  `centralizedPublicOutput`